Applications/VpView/vpViewCore.cxx

```diff
@@ -5622,56 +5625,24 @@ void vpViewCore::createDisplayOutline(vpProject* project)
     {
-    if (project->AOIUpperLeftLatLon[0] != 444)
-      {
-      double ul[] = { project->AOIUpperLeftLatLon[1],
-                      project->AOIUpperLeftLatLon[0], 0.0, 1.0 };
-      this->LatLonToWorldMatrix->MultiplyPoint(ul, ul);
-      ul[0] /= ul[3];
-      ul[1] /= ul[3];
-      if (project->AOILowerRightLatLon[0] != 444)
-        {
-        double lr[] = { project->AOILowerRightLatLon[1],
-                        project->AOILowerRightLatLon[0], 0.0, 1.0 };
-        this->LatLonToWorldMatrix->MultiplyPoint(lr, lr);
-        lr[0] /= lr[3];
-        lr[1] /= lr[3];
-        if (project->AOILowerLeftLatLon[0] != 444 &&
-            project->AOIUpperRightLatLon[0] != 444)
-          {
-          // All 4 corners.
-          double ll[] = { project->AOILowerLeftLatLon[1],
-                          project->AOILowerLeftLatLon[0], 0.0, 1.0 };
-          this->LatLonToWorldMatrix->MultiplyPoint(ll, ll);
-          ll[0] /= ll[3];
-          ll[1] /= ll[3];
-          double ur[] = { project->AOIUpperRightLatLon[1],
-                          project->AOIUpperRightLatLon[0], 0.0, 1.0 };
-          this->LatLonToWorldMatrix->MultiplyPoint(ur, ur);
-          ur[0] /= ur[3];
-          ur[1] /= ur[3];
-          aoiOutlinePoints->SetPoint(0, ll);
-          aoiOutlinePoints->SetPoint(1, lr);
-          aoiOutlinePoints->SetPoint(2, ur);
-          aoiOutlinePoints->SetPoint(3, ul);
-          }
-        else
-          {
-          // Two corners.
-          aoiOutlinePoints->SetPoint(0, ul[0], lr[1], 0);
-          aoiOutlinePoints->SetPoint(1, lr[0], lr[1], 0);
-          aoiOutlinePoints->SetPoint(2, lr[0], ul[1], 0);
-          aoiOutlinePoints->SetPoint(3, ul[0], ul[1], 0);
-          }
-        }
-      else
-        {
-        // Just one corner - show a 'notch' in the upper left corner.
-        aoiOutlinePoints->SetPoint(0, ul[0], ul[1] - 100, 0);
-        aoiOutlinePoints->SetPoint(1, ul[0], ul[1], 0);
-        aoiOutlinePoints->SetPoint(2, ul[0] + 100, ul[1], 0);
-        aoiOutlinePoints->SetPoint(3, ul[0], ul[1], 0);
-        }
+    if (project->AOI.GCS > 0)
+      {
+      const auto& aoi =
+        vgGeodesy::convertGcs(project->AOI, vgGeodesy::LatLon_Wgs84);
+      const auto& corners = aoi.Coordinate;
+      double ul[3] = {corners[0].Easting, corners[0].Northing, 0.0};
+      double ur[3] = {corners[1].Easting, corners[1].Northing, 0.0};
+      double lr[3] = {corners[2].Easting, corners[2].Northing, 0.0};
+      double ll[3] = {corners[3].Easting, corners[3].Northing, 0.0};
+      vtkVgApplyHomography(ul, this->LatLonToWorldMatrix, ul);
+      vtkVgApplyHomography(ur, this->LatLonToWorldMatrix, ur);
+      vtkVgApplyHomography(lr, this->LatLonToWorldMatrix, lr);
+      vtkVgApplyHomography(ll, this->LatLonToWorldMatrix, ll);
+
+      aoiOutlinePoints->SetPoint(0, ll);
+      aoiOutlinePoints->SetPoint(1, lr);
+      aoiOutlinePoints->SetPoint(2, ur);
+      aoiOutlinePoints->SetPoint(3, ul);
```

Comment:
I don't recall why potentially would have two corners (or only one)- but previously supported multiple cases. Not important to continue supporting these other cases? Maybe the other case aren't possible any more?

Reply:
(See reply to previous comment.)